app/rocket/kinetics.py

```diff
@@ -0,0 +1,90 @@
+from threading import Thread
+from collections import deque
+
+import numpy as np
+import logging
+
+WINDOW_SIZE = 50
+
+class Kinetics(Thread):
+    def __init__(self, device_factory):
+        Thread.__init__(self)
+        self.accelerometer = device_factory.accelerometer
+
+        self.time_series = deque(np.arange(WINDOW_SIZE), maxlen=WINDOW_SIZE)
+
+        self.acceleration = TimeWindow()
+        self.velocity = TimeWindow()
+        self.position = TimeWindow()
+        self.active = False
+
+    def activate(self):
+        if self.active:
+            raise Exception('Active Kinetics Model')
```

Comment:
do you guys think we should raise, or just continue silently and ensure not to spin up another thread

Reply:
Probably just continue silently while making sure another thread isn't created - I don't think we need to see the trace output of that on the final release: might be useful until then though.